src/library/colordelegate.cpp

```diff
@@ -0,0 +1,34 @@
+#include "library/colordelegate.h"
+
+#include <QColor>
+#include <QPainter>
+#include <QStyle>
+#include <QTableView>
+
+#include "library/trackmodel.h"
+#include "util/color/rgbcolor.h"
+
+ColorDelegate::ColorDelegate(QTableView* pTableView)
+        : TableItemDelegate(pTableView) {
+}
+
+void ColorDelegate::paintItem(QPainter* painter, const QStyleOptionViewItem& option, const QModelIndex& index) const {
+    DEBUG_ASSERT(index.data().isValid());
+    if (index.data().isNull()) {
+        // Filter out track color that is hidden
+        if (option.state & QStyle::State_Selected) {
+            painter->fillRect(option.rect, option.palette.highlight());
+        }
+        return;
+    }
+
+    QColor color = mixxx::toQColor(mixxx::RgbColor(index.data().toUInt()));
```

Comment:
Slightly more readable and maintainable variant:

```
const auto rgbColor = mixxx::RgbColor::optional(index.data());
painter->fillRect(option.rect, toQColor(rgbColor));
```

It checks for isNull() again, but that doesn't matter. The namespace prefix for toQColor() might be redundant considering [argument dependent lookup](https://en.wikipedia.org/wiki/Argument-dependent_name_lookup).

I would prefer to first do the conversion from QVariant to RgbColor::optional_t at the beginning of the function and then check for `if (!rgbColor) {...}`. First lift the data to the correct abstraction level and then apply the domain logic.